lib/iris/tests/unit/util/test_equalise_attributes.py

```diff
@@ -66,21 +65,21 @@ def _test(self, cubes, expect_attributes, expect_removed):
         # Exercise basic operation
         actual_removed = equalise_attributes(working_cubes)
         # Check they are the same cubes
-        self.assertEqual(working_cubes, original_working_list)
+        assert working_cubes == original_working_list
         # Check resulting attributes all match the expected set
         for cube in working_cubes:
-            self.assertEqual(cube.attributes, expect_attributes)
+            assert cube.attributes == expect_attributes
         # Check removed attributes all match as expected
-        self.assertEqual(len(actual_removed), len(expect_removed))
+        assert len(actual_removed) == len(expect_removed)
         for actual, expect in zip(actual_removed, expect_removed):
-            self.assertEqual(actual, expect)
+            assert all(actual) == all(expect)
```

Comment:
I assume you added `all()` to get it to shut up about comparing iterables. But this isn't doing what you think - it's just checking that everything in `actual`/`expect` is truthy:

```python
>>> all(["a", 1, 2])
True
>>> all(["a", 1, 0])
False
```

Basically: the fact that this is always returning `True` for this test is not evidence that the code is working.

The correct change will depend on what `actual` and `expect` actually are.

Reply:
Yeah that was a brain fart, apologies! They seem to be either a dict or a list, which gives me two different cases to handle is all